### PR TITLE
find_target can no longer choose people on the arrivals shuttle

### LINKED ISSUE #57983: heretic gets targets on arrivals shuttle
Reporting client version: 514.1552

## Round ID:
https://sb.atlantaned.space/rounds/159043
>be me
>heretic, less go
>get target on arrivals shuttle...?
>huh???

## Reproduction:

 be heretic, put someone on arrivals shuttle, select target, profit.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

fixes #57983 

## Why It's Good For The Game

they really shouldn't be picking people on the arrival shuttle as it goes to an area nobody can reach and it SUCKS.

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
What happens if find_target is called and there's no valid targets?

**Comment by tralezab** [comment]
@LemonInTheDark It returns null, and objectives turn into "Free Objective"

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -118,7 +118,8 @@ GLOBAL_LIST_EMPTY(objectives)
 		if(O.late_joiner)
 			try_target_late_joiners = TRUE
 	for(var/datum/mind/possible_target in get_crewmember_minds())
-		if(!(possible_target in owners) && ishuman(possible_target.current) && (possible_target.current.stat != DEAD) && is_unique_objective(possible_target,dupe_search_range))
+		var/target_area = get_area(possible_target.current)
+		if(!(possible_target in owners) && ishuman(possible_target.current) && (possible_target.current.stat != DEAD) && is_unique_objective(possible_target,dupe_search_range) && !istype(target_area, /area/shuttle/arrival))
```
Comment (MrDoomBringer) [inline code/game/gamemodes/objective.dm:122]:
```suggestion
		if(!(possible_target in owners) && ishuman(possible_target.current) && (possible_target.current.stat != DEAD) && is_unique_objective(possible_target,dupe_search_range) && (HAS_TRAIT(SSstation, STATION_TRAIT_LATE_ARRIVALS) || !istype(target_area, /area/shuttle/arrival)))
```
Probably don't want this hitting during the late arrivals station trait.

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]
Also, it'd probably be prudent to expand this massive conditional into a couple early returns at this point.

**Comment by LemonInTheDark** [comment]
isn't there that one station trait that spawns everyone on the shuttle roundstart?  would that fuck with this?

**Comment by MrDoomBringer** [comment]
isn't the point of shivering to heat you up?

**Comment by tralezab** [comment]
i'll split the big check into a bunch of early returns

**Comment by LemonInTheDark** [comment]
> 
> 
> isn't the point of shivering to heat you up?

god damn it

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**